staging/src/k8s.io/apiserver/pkg/authentication/token/cache/cached_token_authenticator_test.go

```diff
@@ -125,3 +130,137 @@ func TestCachedTokenAuthenticatorWithAudiences(t *testing.T) {
 		t.Errorf("Expected user1-different")
 	}
 }
+
+func BenchmarkCachedTokenAuthenticator(b *testing.B) {
+	tokenCount := []int{100, 500, 2500, 12500, 62500}
+	for _, tc := range tokenCount {
+		b.Run(fmt.Sprintf("toks-%v", tc), newSingleBenchmark(tc).bench)
+	}
+}
+
+func newSingleBenchmark(tokenCount int) *singleBenchmark {
+	s := &singleBenchmark{
+		tokenCount: tokenCount,
+	}
+	s.makeTokens()
+	return s
+}
+
+type singleBenchmark struct {
+	tokenCount int
+
+	tokenToResponse map[string]*cacheRecord
+	tokens          []string
+	chokepoint      chan struct{}
+
+	b  *testing.B
+	wg sync.WaitGroup
+}
+
+func (s *singleBenchmark) makeTokens() {
+	s.tokenToResponse = map[string]*cacheRecord{}
+	s.tokens = []string{}
+
+	for i := 0; i < s.tokenCount; i++ {
+		tok := fmt.Sprintf("token_%v", i)
```

Comment:
Can we use more realistic tokens?  Possibly a mix of JWTs and 256 bit random noise tokens?